src/SFA.DAS.AdminService.Web/Validators/Roatp/Applications/RoatpFinancialClarificationViewModelValidator.cs

```diff
@@ -1,93 +1,158 @@
 ﻿using System;
+using System.Collections.Generic;
 using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
 using FluentValidation;
 using FluentValidation.Validators;
+using Microsoft.AspNetCore.Http;
+using SFA.DAS.AdminService.Common.Validation;
 using SFA.DAS.AdminService.Web.ViewModels.Apply.Financial;
 using SFA.DAS.AdminService.Web.ViewModels.Roatp.Financial;
 using SFA.DAS.AssessorService.ApplyTypes.Roatp.Apply;
 
 namespace SFA.DAS.AdminService.Web.Validators.Roatp.Applications
 {
-    public class RoatpFinancialClarificationViewModelValidator : AbstractValidator<RoatpFinancialClarificationViewModel>
+    public class RoatpFinancialClarificationViewModelValidator : IRoatpFinancialClarificationViewModelValidator
     {
-        public RoatpFinancialClarificationViewModelValidator()
+        private const long MaxFileSizeInBytes = 5 * 1024 * 1024;
+        private const string MaxFileSizeExceeded = "The selected file must be smaller than 5MB";
+        private const string FileMustBePdf = "The selected file must be a PDF";
+        private const string ClarificationFile = "ClarificationFile";
+        private const string ClarificationResponse = "ClarificationResponse";
+
+        public async Task<ValidationResponse> Validate(RoatpFinancialClarificationViewModel vm,
+            bool isClarificationFileUpload, bool isClarificationOutcome)
         {
-            RuleFor(vm => vm).Custom((vm, context) =>
+            var validationResponse = new ValidationResponse
+            {
+                Errors = new List<ValidationErrorDetail>()
+            };
+
+            if (isClarificationFileUpload)
+            {
+                if (vm.FilesToUpload != null)
+                {
+                    foreach (var file in vm.FilesToUpload)
+                    {
+
+                        if (!FileContentIsValidForPdfFile(file))
+                        {
+                            validationResponse.Errors.Add(new ValidationErrorDetail(ClarificationFile, FileMustBePdf));
+
+                            break;
+                        }
+                        else if (file.Length > MaxFileSizeInBytes)
+                        {
+                            validationResponse.Errors.Add(new ValidationErrorDetail(ClarificationFile,
+                                MaxFileSizeExceeded));
+                            break;
+                        }
+                    } 
+                    if (vm.FilesToUpload.Count==0)
+                        validationResponse.Errors.Add(new ValidationErrorDetail(ClarificationFile, "Select a file"));
+                }
+            }
```

Comment:
Could this be done as separate method within the validator? The Validate method is quite lengthy otherwise